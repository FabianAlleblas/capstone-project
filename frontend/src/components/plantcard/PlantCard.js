//import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import monsteraimage from '../../assets/plant-images/3.jpg'

export default function PlantCard({ plantname, plantspecies }) {
  PlantCard.propTypes = {
    plantname: PropTypes.string.isRequired,
    plantspecies: PropTypes.string.isRequired,
  }

  return (
    <CardWrapper>
      <ImgContainer src={monsteraimage} />
      <PlantName>{plantname}</PlantName>
      <PlantSpecies>{plantspecies}</PlantSpecies>
    </CardWrapper>
  )
}

const CardWrapper = styled.div`
  display: grid;
  justify-items: center;
`

const ImgContainer = styled.div`
  background-image: url(${(props) => props.src});
  background-position: center;
  background-size: cover, contain;
  border-radius: 50%;
  height: 100px;
  margin-bottom: 10px;
  width: 100px;
`

const PlantName = styled.p`
  color: var(--primary-plant-font-color);
  font-size: 1.25rem;
`

const PlantSpecies = styled.p`
  color: var(--secondary-plant-font-color);
  font-size: 0.75rem;
`
