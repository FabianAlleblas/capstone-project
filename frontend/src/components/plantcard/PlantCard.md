```js
<PlantCard plantname={'Jack'} plantspecies={'Monstera deliciosa'} />
```
