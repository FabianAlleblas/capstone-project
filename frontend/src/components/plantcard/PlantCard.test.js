import { render } from '@testing-library/react'
import 'jest-styled-components'
import PlantCard from './PlantCard'

describe('PlantCard', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(
      <PlantCard plantname={'Jack'} plantspecies={'Monstera deliciosa'} />
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/Jack/i)).toBeInTheDocument()
    expect(getByText(/Monstera deliciosa/i)).toBeInTheDocument()
  })
})
