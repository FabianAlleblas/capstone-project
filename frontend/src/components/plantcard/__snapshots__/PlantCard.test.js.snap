// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantCard renders correctly 1`] = `
.c0 {
  display: grid;
  justify-items: center;
}

.c1 {
  background-image: url(3.jpg);
  background-position: center;
  background-size: cover,contain;
  border-radius: 50%;
  height: 100px;
  margin-bottom: 10px;
  width: 100px;
}

.c2 {
  color: var(--primary-plant-font-color);
  font-size: 1.25rem;
}

.c3 {
  color: var(--secondary-plant-font-color);
  font-size: 0.75rem;
}

<div
  class="c0"
>
  <div
    class="c1"
    src="3.jpg"
  />
  <p
    class="c2"
  >
    Jack
  </p>
  <p
    class="c3"
  >
    Monstera deliciosa
  </p>
</div>
`;
